lib/stencil-push.utils.js

```diff
@@ -311,7 +311,16 @@ utils.getVariations = async (options) => {
         return { ...options, variationId: foundVariation.uuid };
     }
     if (activate) {
-        return { ...options, variationId: variations[0].uuid };
+        const indexofVariationToActivate = variations.findIndex(
```

Comment:
I think we can simplify to (check the condition above) 
`const foundVariation = variations.find((item) => item.name === activate);`
and 
`return { ...options, variationId: foundVariation.uuid };`

Reply:
Also I don't think we need the previous condition 
`if (!activate && activate !== undefined) {`
it will work only if activate is _0_ or _""_ or _false_

Reply:
Yeah, I actually suggested these changes in the provided fixed code above)

Reply:
@MaxGenash Agree. That's the better way